src/common/pf_bg_worker.c

```diff
@@ -0,0 +1,127 @@
+
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2032 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "pf_includes.h"
+
+#include <inttypes.h>
+
+/* Events handled by bg worker task */
+
+#define BG_JOB_EVENT_UPDATE_PORTS_STATUS  BIT (0)
+#define BG_JOB_EVENT_SAVE_ASE_NVM_DATA    BIT (1)
+#define BG_JOB_EVENT_SAVE_IM_NVM_DATA     BIT (2)
+#define BG_JOB_EVENT_SAVE_PDPORT_NVM_DATA BIT (3)
+
+static void bg_worker_task (void * arg);
+
+void pf_bg_worker_init (pnet_t * net)
+{
+   net->pf_bg_worker.events = os_event_create();
+   CC_ASSERT (net->pf_bg_worker.events != NULL);
+
+   os_thread_create (
+      "p-net_bg_worker",
+      net->fspm_cfg.pnal_cfg.bg_worker_thread.prio,
+      net->fspm_cfg.pnal_cfg.bg_worker_thread.stack_size,
+      bg_worker_task,
+      (void *)net);
+}
+
+/**
+ * Task for running in a background thread
+ *
+ * @param arg              InOut: Thread argument, should be pnet_t *
+ */
+int pf_bg_worker_start_job (pnet_t * net, pf_bg_job_t job_id)
+{
+   if (net == NULL)
+   {
+      return -1;
+   }
+
+   switch (job_id)
+   {
+
```

Comment:
Make sure to run clang-format. Should there be an empty line here?

Reply:
OK, removed.
clang-format did not remove is automatically